sdk/Sdk.Generators/FunctionMetadataProviderGenerator.cs

```diff
@@ -0,0 +1,384 @@
+                IDictionary<string, object> attributeProperties = GetAttributeProperties(attribMethodSymbol, attributeData);
+
+                string attributeName = attributeData.AttributeClass!.Name;
+                string bindingName = parameter.Identifier.ValueText;
+
+                // properly format binding types by removing "Attribute" and "Input" descriptors
+                string bindingType = attributeName.Replace("Attribute", "");
+                bindingType = bindingType.Replace("Input", "");
+
+                // Set binding direction
+                string bindingDirection = "In";
+                if (parameterSymbol.Type is INamedTypeSymbol parameterNamedType &&
+                    parameterNamedType.IsGenericType &&
+                    parameterNamedType.ConstructUnboundGenericType().ToString() == "Microsoft.Azure.Functions.DotNetWorker.OutputBinding<>")
+                {
+                    bindingDirection = "Out";
+                }
+
+                // Create raw binding anonymous type, example:
+                /*  var binding1 = new {
+                    name = "req",
+                    type = "HttpTrigger",
+                    direction = "In",
+                    authLevel = Enum.GetName(typeof(AuthorizationLevel),0),
+                    methods = new List<string> { "get","post" },
+                };*/
+                indentedTextWriter.WriteLine("var " + functionName + "Binding" + bindingCount.ToString() + " = new {"); // give each binding unique name of functionName + "binding" + number
+                indentedTextWriter.Indent++;
+                indentedTextWriter.WriteLine("name = \"" + bindingName + "\",");
+                indentedTextWriter.WriteLine("type = \"" + bindingType + "\",");
+                indentedTextWriter.WriteLine("direction = \"" + bindingDirection + "\",");
+
+                // Add additional bindingInfo to the anonymous type because some functions have more properties than others
+                foreach (var prop in attributeProperties)
+                {
+                    var propertyName = prop.Key;
+
+                    if (prop.Value.GetType().IsArray)
+                    {
+                        string arr = FormatArray((IEnumerable)prop.Value);
+                        indentedTextWriter.WriteLine(propertyName + " = " + arr + ",");
+                    }
+                    else
+                    {
+                        var propertyValue = FormatObject(prop.Value);
+                        indentedTextWriter.WriteLine(propertyName + " = " + propertyValue + ",");
+                    }
+                }
+
+                indentedTextWriter.Indent--;
+                indentedTextWriter.WriteLine("};");
+                indentedTextWriter.WriteLine("var " + functionName + "Binding" + bindingCount.ToString() + "JSONstring = JsonSerializer.Serialize(" + functionName + "Binding" + bindingCount.ToString() + ").ToString();");
```

Comment:
nit: string interpolation might be easier to read here. See other places as well.